From marcin.slusarz@gmail.com  Fri Jul 10 16:05:52 2009
From: Marcin Slusarz <marcin.slusarz@gmail.com>
Date: Wed, 08 Jul 2009 22:07:20 +0200
Subject: Staging: rt*: fix wait_queue_head_t declaration on stack
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Message-ID: <4A54FC78.2010101@gmail.com>


From: Marcin Slusarz <marcin.slusarz@gmail.com>

othwerwise lockdep complains:
"INFO: trying to register non-static key.
the code is fine but needs lockdep annotation.
turning off the locking correctness validator."

Signed-off-by: Marcin Slusarz <marcin.slusarz@gmail.com>
Acked-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/rt_main_dev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rt2860/rt_main_dev.c
+++ b/drivers/staging/rt2860/rt_main_dev.c
@@ -190,7 +190,7 @@ int rt28xx_close(IN PNET_DEV dev)
 	BOOLEAN 		Cancelled = FALSE;
 	UINT32			i = 0;
 #ifdef RT2870
-	DECLARE_WAIT_QUEUE_HEAD(unlink_wakeup);
+	DECLARE_WAIT_QUEUE_HEAD_ONSTACK(unlink_wakeup);
 	DECLARE_WAITQUEUE(wait, current);
 
 	//RTMP_SET_FLAG(pAd, fRTMP_ADAPTER_REMOVE_IN_PROGRESS);
